sklearn/linear_model/tests/test_base.py

```diff
@@ -44,26 +45,28 @@ def test_linear_regression():
 
 def test_linear_regression_sample_weights():
     rng = np.random.RandomState(0)
+    n_samples, n_features = 6, 50  # over-determined system
 
-    for n_samples, n_features in ((6, 5), (5, 10)):
+    for fit_intercept in [True, False]:
         y = rng.randn(n_samples)
         X = rng.randn(n_samples, n_features)
         sample_weight = 1.0 + rng.rand(n_samples)
 
-        clf = LinearRegression()
-        clf.fit(X, y, sample_weight)
-        coefs1 = clf.coef_
+        reg = LinearRegression(fit_intercept=fit_intercept)
+        reg.fit(X, y, sample_weight=sample_weight)
+        coefs1 = reg.coef_
 
-        assert_equal(clf.coef_.shape, (X.shape[1], ))
-        assert_greater(clf.score(X, y), 0.9)
-        assert_array_almost_equal(clf.predict(X), y)
+        assert_equal(reg.coef_.shape, (X.shape[1], ))
+        assert_greater(reg.score(X, y, sample_weight=sample_weight), 0.9)
+        assert_greater(r2_score(y, reg.predict(X),
+                       sample_weight=sample_weight), 0.9)  # same as above
 
         # Sample weight can be implemented via a simple rescaling
         # for the square loss.
         scaled_y = y * np.sqrt(sample_weight)
         scaled_X = X * np.sqrt(sample_weight)[:, np.newaxis]
-        clf.fit(X, y)
-        coefs2 = clf.coef_
+        reg.fit(scaled_X, scaled_y)
+        coefs2 = reg.coef_
 
         assert_array_almost_equal(coefs1, coefs2)
```

Comment:
`intercept_` looks good. I am not sure to which `x2` you refer to. Do you mean to multiply all by 2 as if `all samples_weight=4`?